lib/widgets/sample_viewer_page.dart

```diff
@@ -132,22 +132,41 @@ class _SampleViewerPageState extends State<SampleViewerPage> {
   }
 
   void onSearchChanged(String searchText) {
-    final List<Sample> results;
+    var results = <Sample>[];
+    // restore the initial list of samples if the search text is empty
     if (searchText.isEmpty) {
-      results = _allSamples;
+      if (widget.category == null) {
+        results = [];
+      } else if (widget.category == SampleCategory.all) {
+        results = _allSamples;
+      } else if (widget.category != null) {
```

Comment:
This can just be `} else {` (with no `if` condition) because the case where `widget.category == null` was already handled in the first branch of the `if` statement. So we know it can't be `null` here.